dashboard/dashboard.py

```diff
@@ -242,17 +252,39 @@ def analyze_experiment(info, experiments_dir, tmp_data_dir,
 
     status = validate_status(tmp_analysis_dir)
 
+    # collect data to dump to data_*.json
+    dump_data = {}
     # read the analyzed data, append a timestamp field, and copy over to the permanent data dir
     if status['success']:
         data_exists = check_file_exists(tmp_analysis_dir, 'data.json')
         if not data_exists:
             status = {'success': False, 'message': 'No data.json file produced by {}'.format(exp_name)}
         else:
             data = read_json(tmp_analysis_dir, 'data.json')
-            data['timestamp'] = date_str
-            data['tvm_hash'] = tvm_hash
-            write_json(analyzed_data_dir, 'data_{}.json'.format(date_str), data)
-
+            data.update({
+                'timestamp'  : date_str,
+                'tvm_hash'   : tvm_hash,
+            })
+            dump_data.update(data)
+
+    # fetch time spent on the experiment
+    run_status = validate_json(info.exp_status_dir(exp_name), 
+                            'start_time', 
+                            'end_time', 
+                            'time_delta', filename='run.json')
+    # only process valid files
+    keys = run_status.keys()
+    if keys and functools.reduce(lambda x, y: x and y, 
+                                     map(lambda x: x in keys, 
+                                        ('start_time', 'end_time', 'time_delta'))):
+        rs_get = run_status.get
+        dump_data.update({
+            'start_time' : rs_get('start_time'),
+            'end_time'   : rs_get('end_time'),
+            'time_delta' : rs_get('time_delta')
+        })
+    # dump date iff we have something to write 
+    write_json(analyzed_data_dir, 'data_{}.json'.format(date_str), dump_data) if len(dump_data) else 'NO DATA; PASS'
```

Comment:
Nitpick: I would prefer there to just be an `if` statement (i.e., `if dump_data: write_json(...)`) instead of having a weird string dangling about